zerver/tests/test_push_notifications.py

```diff
@@ -1490,6 +1495,166 @@ def test_register_push_device_error(self) -> None:
         self.assert_json_error(result, "Invalid encrypted_push_registration")
 
 
+class RegisterPushDeviceToServer(ZulipTestCase):
```

Comment:
Consider putting these tests in a new `test_push_registration.py` too; I think it'll be net helpful to have a dedicated test file for that long term. (We don't need to move existing tests right now; that can be a future pass)